pkg/services/searchV2/extender.go

```diff
@@ -0,0 +1,48 @@
+package searchV2
+
+import (
+	"github.com/blugelabs/bluge"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+type ExtendDashboardFunc func(uid string, doc *bluge.Document) error
+type FramerFunc func(field string, value []byte) bool
+
+type QueryExtender interface {
+	GetFramer(frame *data.Frame) FramerFunc
+}
+
+type DocumentExtender interface {
+	GetDashboardExtender(orgID int64, uids []string) ExtendDashboardFunc
+}
+
+type DashboardIndexExtender interface {
+	GetDocumentExtender() DocumentExtender
```

Comment:
@FZambia i added `GetDocumentExtender` to match your `GetQueryExtender` change and implemented them both in the enterprise PR